### PR TITLE
[Backend] Fix parameter patching

### PR DESCRIPTION
**Pull request by numerology**
Previous patching does not work for the following codepath:

In SDK use placeholder in parameter default value → In SDK create_run_from_pipeline_func/package → Run triggered and failed 

Change to patch RunDetail and WorkflowSpec in resource manager.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3145)

### CODE REVIEW

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -283,6 +286,32 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 	}
 	// Append provided parameter
 	workflow.OverrideParameters(parameters)
+	// Patch the default value to workflow spec.
+	if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+		patchedSlice := make([]workflowapi.Parameter, 0)
```
Comment (IronPan) [inline backend/src/apiserver/resource/resource_manager.go]: could u extract this to a function for readability similar to OverrideParameters() in previous line.

Comment (numerology) [reply backend/src/apiserver/resource/resource_manager.go]: SG. Done.

**Comment by numerology** [comment]
Manually merging b/c Yang is OOO today.

**Pull request merged**